File: vue.jsonforms/src/entities/Affiliate/Affiliate.ts
import { ProviderRepresentative } from "@/entities/providers-and-introducers/ProviderRepresentative";
import { ProvidersProductDetails } from "@/entities/providers-and-introducers/ProvidersProductDetails";
import { AffiliateDetails } from "@/entities/Affiliate/AffiliateDetails";
import { ProvidersAndIntroducersBase } from "../providers-and-introducers/ProvidersAndIntroducersBase";

export class Affiliate extends ProvidersAndIntroducersBase {
  public id: string = "";
  public details: AffiliateDetails = new AffiliateDetails();
  public marketingProviderDetails: AffiliateDetails = new AffiliateDetails();
  public representative: ProviderRepresentative = new ProviderRepresentative();
  public products: ProvidersProductDetails[] = [] as ProvidersProductDetails[];
  public isAffiliateMarketingProvider: boolean = false;

  public status: string = "";
  public tasks: string = "";
}

File: CSharp.Functions/Common/Entities/Affiliate.cs
namespace Common.Entities
{
    public class Affiliate : ProvidersAndIntroducersBase
    {
        public string? Id { get; set; }
        public AffiliateDetails? Details { get; set; }
        public AffiliateDetails? MarketingProviderDetails { get; set; }
        public bool IsAffiliateMarketingProvider { get; set; } = false;
        public ProviderRepresentative? Representative { get; set; }
        public string OnboardingType { get; set; } = OnboardingTypes.Affiliate.ToString();
    }
}
